jib-core/src/main/java/com/google/cloud/tools/jib/configuration/BuildConfiguration.java

```diff
@@ -302,4 +302,20 @@ public boolean getAllowInsecureRegistries() {
   public ImmutableList<LayerConfiguration> getLayerConfigurations() {
     return layerConfigurations;
   }
+
+  /**
+   * Creates a new {@link RegistryClient.Factory} and sets fields from the build configuration.
+   *
+   * @param imageConfiguration the {@link ImageConfiguration} to use for the server URL and image
+   *     name
+   * @return a new {@link RegistryClient.Factory}
+   */
+  public RegistryClient.Factory newRegistryClientFactory(ImageConfiguration imageConfiguration) {
```

Comment:
It seems odd for BuildConfiguration, which is essentially a value object that encodes the intent, now knows how to create RegistryClients, but has to have part of itself passed back in.

Reply:
How should we select which `ImageConfiguration` we're using to create the `RegistryClient`?

Reply:
>But RegistryClient already has a factory(), and I don't think another level of indirection is helpful.

In this case maybe RegistryClient should have a newFactory method that takes a build configuration?

Reply:
Although maybe we don't want to require `BuildConfiguration` this deep down?

Reply:
Yea, the `registry` package is definitely lower-level than the `configuration` package (which is only "seen" by the `builder` package and higher).

Reply:
I ended up adding a separate method for the target image and base image; I'm not sure if the method fits on `ImageConfiguration` considering it pulls a few other values from `BuildConfiguration`.

Reply:
Yea, if the method goes into `ImageConfiguration`, it shouldn't directly take a `BuildConfiguration` since `ImageConfiguration` should not depend on `BuildConfiguration`.